lib/ar-find-in-batches-with-order.rb

```diff
@@ -30,27 +27,27 @@ def find_in_batches_with_order(options = {})
 
         break if records_size < batch_size
 
-        start = records.last.try(property_key)
+        next_start = records.last.try(property_key)
+        with_start_ids.clear if start != next_start
+        start = next_start
 
-        records = strict_mode ? (direction == :desc ? relation.where("#{sanitized_key} <= ?", start).to_a : relation.where("#{sanitized_key} >= ?", start).to_a) : (direction == :desc ? relation.where("#{sanitized_key} < ?", start).to_a : relation.where("#{sanitized_key} > ?", start).to_a)
+        records.each do |record|
```

Comment:
What you think of a more functional approach a la 

```
with_start_ids = records.filter { |record| record.try(property_key) == start }.map(&:id)
```

Reply:
it needs to maintain the list of dups from the previous batch so we can't re-assign it here